src/workerd/jsg/ser.c++

```diff
@@ -43,6 +47,42 @@ void Serializer::ThrowDataCloneError(v8::Local<v8::String> message) {
   isolate->ThrowException(makeDOMException(isolate, message, "DataCloneError"));
 }
 
+v8::Maybe<bool> Serializer::WriteHostObject(v8::Isolate* isolate, v8::Local<v8::Object> object) {
+  try {
+    if (object->InternalFieldCount() != Wrappable::INTERNAL_FIELD_COUNT ||
+        object->GetAlignedPointerFromInternalField(Wrappable::WRAPPABLE_TAG_FIELD_INDEX) !=
+            &Wrappable::WRAPPABLE_TAG) {
+      // v8::ValueSerializer by default will send us anything that has internal fields, but this
+      // object doesn't appear to match the internal fields expected on a JSG object.
+      JSG_FAIL_REQUIRE(DOMDataCloneError, "Foreign host object could not be serialized.");
```

Comment:
non-blocking nit: Hmm... not sure if this is a great user-facing error message. They might not be aware of what a "Foreign host object" is. Fortunately this case shouldn't ever actually happen so it's probably ok.

Reply:
I also don't know what a "foreign host object" is, unfortunately! (I don't know what kinds of objects V8 puts internal fields on, though I'm pretty sure there is at least one type that it does this with.)

Reply:
heh, maybe just leave the `Foreign host` part of the message off then? non-blocking either way.

Reply:
I figure if someone reports the error to us, the words can at least help us figure out where to look.